fiksi/src/lib.rs

```diff
@@ -487,6 +500,14 @@ impl System {
                     }
                 }
 
+                if opts.perturb {
+                    for free_variable in free_variables.iter().copied() {
+                        let variable = &mut self.variables[free_variable as usize];
+                        *variable += *variable * (1. / 8196.) * rng.next_f64()
+                            + (1. / 65568.) * rng.next_f64();
+                    }
+                }
```

Comment:
This adds a random perturbation relative to the magnitude of the variable, plus a small perturbation regardless of magnitude (when the value is 0, for example, it still needs to be perturbed). I don't entirely like the perturbation that is insensitive to magnitude. It's not great if a circle radius becomes negative, for example, or in case the system has very small values in general, this perturbation may relatively be huge.